File: api/routes/drive_sync/cover_update.py
"""Cover update endpoints — proxy to BedReadDriveSync."""

from __future__ import annotations

from fastapi import APIRouter
from fastapi.responses import JSONResponse

from api.routes.drive_sync.proxy import drive_get, drive_post

router = APIRouter(prefix="/cover-update", tags=["Drive Sync"])


async def _proxy_get(path: str, params: dict | None = None, timeout: float = 60.0) -> JSONResponse:
    return await drive_get(path, params=params, timeout=timeout)


async def _proxy_post(path: str, json_body: dict | None = None, params: dict | None = None) -> JSONResponse:
    return await drive_post(path, json_body=json_body, params=params, timeout=120.0)


@router.get("/check-all")
async def check_all(cover_filename: str = "cover1.jpg") -> JSONResponse:
    return await _proxy_get("/api/drive-sync/cover-update/check-all", params={"cover_filename": cover_filename}, timeout=120.0)


@router.get("/check-updated")
async def check_updated() -> JSONResponse:
    return await _proxy_get("/api/drive-sync/cover-update/check-updated")


@router.post("/upload/{folder_id}/{story_id}")
async def upload_cover(folder_id: str, story_id: str, cover_filename: str = "cover1.jpg") -> JSONResponse:
    return await _proxy_post(
        f"/api/drive-sync/cover-update/upload/{folder_id}/{story_id}",
        params={"cover_filename": cover_filename}
    )

File: CreateStory_FE/src/api/BedReadDriveSync/coverUpdate.ts
import { apiFetch } from '../client';
import type {
  CheckAllResponse,
  CheckUpdatedResponse,
  CoverUpdateUploadResponse,
} from '../types';

export async function checkCoverUpdateAll(coverFilename: string = 'cover1.jpg'): Promise<CheckAllResponse> {
  return apiFetch<CheckAllResponse>(
    `/api/drive-sync/cover-update/check-all?cover_filename=${encodeURIComponent(coverFilename)}`,
    { timeout: 300000 }
  );
}

export async function checkCoverUpdateUpdated(): Promise<CheckUpdatedResponse> {
  return apiFetch<CheckUpdatedResponse>('/api/drive-sync/cover-update/check-updated', { timeout: 30000 });
}

export async function uploadCoverUpdate(
  folderId: string,
  storyId: string,
  coverFilename: string = 'cover1.jpg'
): Promise<CoverUpdateUploadResponse> {
  return apiFetch<CoverUpdateUploadResponse>(
    `/api/drive-sync/cover-update/upload/${encodeURIComponent(folderId)}/${encodeURIComponent(storyId)}?cover_filename=${encodeURIComponent(coverFilename)}`,
    { method: 'POST', timeout: 120000 }
  );
}
